src/providers/TokenProvider.tsx

```diff
@@ -184,10 +208,66 @@ export const TokenProvider = ({ children }: TokenProps) => {
     );
   };
 
+  const loadTransferAndApprovalLogs = useCallback(async () => {
+    if (!tokenSelected || !address) return;
+
+    const transferLogs = await customClient.publicClient.getLogs({
+      address: tokenSelected.address,
+      event: WONDER_TOKEN_EVENT_TRANSFER_ABI,
+      args: { from: address },
+      fromBlock: 'earliest',
+      toBlock: 'latest',
+    });
+
+    const approvalLogs = await customClient.publicClient.getLogs({
+      address: tokenSelected.address,
+      event: WONDER_TOKEN_EVENT_APPROVAL_ABI,
+      args: { owner: address },
+      fromBlock: 'earliest',
+      toBlock: 'latest',
+    });
+
+    const tmpLogs = [...transferLogs, ...approvalLogs];
+
+    const sortedLogs = tmpLogs.sort((a, b) => {
```

Comment:
I think you could try using a generic type `const sortedLogs = tmpLogs.sort<EventLogs>((a, b) => {` and `setLogs(sortedLogs);`

Reply:
Doesn't seem to work. 
The typing here is just to match the expected object in the UI (ex. log.args.spender), that's why it ended with an unknown and a custom type.
I tried using the native type Log<typeof ABI_EVENT> and the GetLogsReturnType, getting nothing but TS mismatch hell.

![Screenshot 2024-09-05 at 9 15 12 PM](https://github.com/user-attachments/assets/64481f6b-493b-4571-91a3-7776434488bb)